.evergreen/config.yml

```diff
@@ -411,71 +427,69 @@ axes:
   - id: "spark"
     display_name: "Spark"
     values:
-      - id: "3.1.2"
-        display_name: "Spark 3.1.2"
+      - id: "3.1.3"
+        display_name: "Spark 3.1.3"
+        variables:
+          SPARK: "3.1.3"
+      - id: "3.2.4"
+        display_name: "Spark 3.2.4"
+        variables:
+          SPARK: "3.2.4"
+      - id: "3.3.4"
+        display_name: "Spark 3.3.4"
         variables:
-          SPARK: "3.1.2"
-      - id: "3.2.2"
-        display_name: "Spark 3.2.2"
+          SPARK: "3.3.4"
+      - id: "3.4.2"
+        display_name: "Spark 3.4.2"
         variables:
-          SPARK: "3.2.2"
+          SPARK: "3.4.2"
+      - id: "3.5.1"
+        display_name: "Spark 3.5.1"
+        variables:
+          SPARK: "3.5.1"
 
 buildvariants:
 
   - matrix_name: "static-checks"
     matrix_spec: { scala: "*", spark: "*" }
     display_name: "Static checks: ${scala} ${spark}"
-    exclude_spec:
-      - {scala: "2.12", spark: "3.2.2"}
-      - {scala: "2.13", spark: "3.1.2"}
-    run_on:
-      - "ubuntu1804-test"
+    <<: *exclude-spec
     tags: ["static-check"]
     tasks:
       - name: "static-checks-task"
 
   - matrix_name: "unit-tests"
     matrix_spec: { scala: "*", spark: "*" }
-    exclude_spec: {scala: "2.13", spark: "3.1.2"}
+    <<: *exclude-spec
```

Comment:
Should static checks and unit tests run on all spark versions? Seems odd that the results would be different (especially for the static checks)

Reply:
The exclude spec definition has the info:

> # Scala 2.12 only works with 3.1.3 spark
> # Scala 2.13 doesn't work on 3.1.3 spark

So we exclude versions that would not compile

Reply:
I'm asking the opposite question, actually.  I understand the exclusions, but I'm not clear why we would need to run static checks more than once (e.g. on Java driver we run static-analysis tests only with Java 17, not 8, 11 and 17).

Reply:
Because different versions of Scala run different code paths (there is separate code for Scalal 2.12 and 2.13 in the connector. 

There are also differences in versions of Spark that we compile against.  Eg Spark 3.4 added the `dropDatabase` change and Spark 3.5 added the encoder changes.  

The unit tests and static checks cover those code paths.